### PR TITLE
GlobalExceptionHandler

### PR DESCRIPTION
**Pull request by oleksii-sukhenko**
add CustomGlobalExceptionHandler class
add meaningful validation for existing DTO classes

### CODE REVIEW

File: src/main/java/mate/academy/controller/BookController.java [hunk 1]
```diff
@@ -34,7 +35,7 @@ public BookDto getBookById(@PathVariable Long id) {
     }
 
     @PostMapping
-    public BookDto createBook(@RequestBody CreateBookRequestDto bookDto) {
+    public BookDto createBook(@RequestBody @Valid CreateBookRequestDto bookDto) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/controller/BookController.java:38]: add `@Valid` in updateBookById endpoint

File: src/main/java/mate/academy/dto/CreateBookRequestDto.java [hunk 2]
```diff
@@ -1,16 +1,25 @@
...
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
 import java.math.BigDecimal;
 import lombok.Getter;
 import lombok.Setter;
 
 @Getter
 @Setter
 public class CreateBookRequestDto {
+    @NotNull
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/dto/CreateBookRequestDto.java]:
```suggestion
    @NotBlank
```

for String

File: src/main/java/mate/academy/dto/CreateBookRequestDto.java [hunk 3]
```diff
@@ -1,16 +1,25 @@
...
     private String title;
+    @NotNull
     private String author;
+    @NotNull
     private String isbn;
+    @NotNull
+    @Min(1)
     private BigDecimal price;
+    @NotNull
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/dto/CreateBookRequestDto.java]:
```suggestion
```

You can validate Size

File: src/main/java/mate/academy/dto/CreateBookRequestDto.java [hunk 4]
```diff
@@ -1,16 +1,25 @@
...
     private String description;
+    @NotNull
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/dto/CreateBookRequestDto.java]:
```suggestion
```
You can validate Size

File: src/main/java/mate/academy/exception/CustomGlobalExceptionHandler.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+            WebRequest request
+    ) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now());
+        body.put("status", HttpStatus.BAD_REQUEST);
+        List<String> errors = ex.getBindingResult().getAllErrors().stream()
+                .map(this::getErrorMessage)
+                .toList();
+        body.put("errors", errors);
+        return new ResponseEntity<>(body, headers, status);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/exception/CustomGlobalExceptionHandler.java]:
```suggestion
        return new ResponseEntity<>(body, headers, HttpStatus.BAD_REQUEST);
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/mate/academy/dto/CreateBookRequestDto.java [hunk 6]
```diff
@@ -1,16 +1,27 @@
...
 @Setter
 public class CreateBookRequestDto {
+    @NotBlank
     private String title;
+    @NotBlank
     private String author;
+    @NotBlank
     private String isbn;
+    @NotNull
+    @Min(1)
```
Comment (okuzan) [inline src/main/java/mate/academy/dto/CreateBookRequestDto.java]: You can use DecimalMin

**Review by okuzan — APPROVED** [review]

**Pull request merged**